plugins/utils/auto-save.nix

```diff
@@ -40,105 +72,109 @@ in
             '';
           }
           ''
-            The message to print en save.
+            The message to print on save.
             This can be a lua function that returns a string.
           '';
 
       dim = helpers.defaultNullOpts.mkNullable (types.numbers.between 0
         1
       ) 0.18 "Dim the color of `message`.";
 
-      cleaningInterval = helpers.defaultNullOpts.mkInt 1250 ''
+      cleaning_interval = helpers.defaultNullOpts.mkUnsignedInt 1250 ''
         Time (in milliseconds) to wait before automatically cleaning MsgArea after displaying
         `message`.
         See `:h MsgArea`.
       '';
     };
 
-    triggerEvents =
-      helpers.defaultNullOpts.mkListOf types.str
-        [
-          "InsertLeave"
-          "TextChanged"
-        ]
-        ''
-          Vim events that trigger auto-save.
-          See `:h events`.
-        '';
-
-    condition =
-      helpers.defaultNullOpts.mkLuaFn
-        ''
-          function(buf)
-            local fn = vim.fn
-            local utils = require("auto-save.utils.data")
-
-            if
-              fn.getbufvar(buf, "&modifiable") == 1 and utils.not_in(fn.getbufvar(buf, "&filetype"), {}) then
-              return true -- met condition(s), can save
-            end
-            return false -- can't save
+    trigger_events = {
+      immediate_save =
+        helpers.defaultNullOpts.mkListOf types.str
+          [
+            "BufLeave"
+            "FocusLost"
+          ]
+          ''
+            Vim events that trigger an immediate save.
+            See `:h events` for events description.
```

Comment:
Just an FYI in case you're not aware: this will (usually) render as one line of text. That's fine if the linebreak is only there to make the code more readable and not needed in the actual docs.

If you _did_ want a "real" line break, markdown supports two kings.

A line break\
like this\
is done by ending the line with a `\`:

```md
A line break\
like this\
is done by ending the line with a `\`:
```

Or a full paragraph break

is done by including one-or-more blank lines:

```md
Or a full paragraph break

is done by including one-or-more blank lines:
```

Reply:
Hmm. Thanks for the information. I didn't know about `\`.
It'll be better with line breaks, of course.

Reply:
I think it's fine as-is, just wanted to make sure it was a deliberate decision either way.

Most of our descriptions reserve line breaks for fairly significant, well, breaks